### ISSUE TITLE
[SVT-AV1 UnitTests] Disable failed unit test cases

### ISSUE THREAD

**Issue by Cidana-Emma** [Cidana-Developers/UnitTests-Task-manager#71]
To do:
- [x] fully run the unit test to collect the failed cases
- [x] Submit the PR in SVT-AV1 Github to disable the failed unit tests
- [x] list dwon test cases with runtime
- [ ] provide a sample code for filtering the unit tests by name
- [x] crashes in transform unit tests
- [x] proposal to use SHARDS (8) for unit test runtime optimization
- [x] new failure cases in SAD

**Comment by Cidana-Emma** [comment]
@Richard-li Please list the test condition as well, the following example for your reference. Thank you!

Test Condition:
1.   test with release version
2.   single thread test for count time of each testcase
2.   Sharded test: 2 sharded for optimized time (Travis provide 2 cores)
3.   test platform: Linux
4.   CPU: i7 8700
5.   Memory: 16G

Total:   13900 tests, 22 tests failed as the following:

**Comment by Richard-li** [comment]
Cidana-Developers / SVT-AV1 2020/01/15 Lastest Source Code Running Result [log_20200115.txt](https://github.com/Cidana-Developers/Cidana-Task-Manager/files/4069141/log_20200115.txt)
Running Time (ALL): 7549885 ms (~= 2.09 h)
- Test with release version
- Single thread test for count time of each testcase
- Test Platform: Linux
- CPU: i7-8700
- Memory: 16 G
- Total 29095 tests, 73 failed as following


[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/0, where GetParam() = (4, 4)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/1, where GetParam() = (4, 8)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/2, where GetParam() = (4, 16)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/3, where GetParam() = (8, 4)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/4, where GetParam() = (8, 8)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/5, where GetParam() = (8, 16)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/6, where GetParam() = (8, 32)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/7, where GetParam() = (16, 4)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/8, where GetParam() = (16, 8)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/9, where GetParam() = (16, 16)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/10, where GetParam() = (16, 32)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/11, where GetParam() = (16, 64)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/12, where GetParam() = (32, 8)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/13, where GetParam() = (32, 16)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/14, where GetParam() = (32, 32)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/15, where GetParam() = (32, 64)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/16, where GetParam() = (64, 16)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/17, where GetParam() = (64, 32)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/18, where GetParam() = (64, 64)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/19, where GetParam() = (64, 128)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/20, where GetParam() = (128, 64)
[  FAILED  ] HBDVarianceTest/HBDVarianceTest.MatchTest/21, where GetParam() = (128, 128)
[  FAILED  ] AVX2_COPY/AV1HbdJntConvolve2DTest.MatchTest/0, where GetParam() = (8, 0, 0, 0, 0)
[  FAILED  ] AVX2_COPY/AV1HbdJntConvolve2DTest.MatchTest/1, where GetParam() = (8, 0, 0, 0, 1)
[  FAILED  ] AVX2_COPY/AV1HbdJntConvolve2DTest.MatchTest/16, where GetParam() = (8, 0, 0, 0, 16)
[  FAILED  ] AVX2_COPY/AV1HbdJntConvolve2DTest.MatchTest/22, where GetParam() = (10, 0, 0, 0, 0)
[  FAILED  ] AVX2_COPY/AV1HbdJntConvolve2DTest.MatchTest/23, where GetParam() = (10, 0, 0, 0, 1)
[  FAILED  ] AVX2_COPY/AV1HbdJntConvolve2DTest.MatchTest/38, where GetParam() = (10, 0, 0, 0, 16)
[  FAILED  ] AVX2_COPY/AV1HbdJntConvolve2DTest.MatchTest/44, where GetParam() = (12, 0, 0, 0, 0)
[  FAILED  ] AVX2_COPY/AV1HbdJntConvolve2DTest.MatchTest/45, where GetParam() = (12, 0, 0, 0, 1)
[  FAILED  ] AVX2_COPY/AV1HbdJntConvolve2DTest.MatchTest/60, where GetParam() = (12, 0, 0, 0, 16)
[  FAILED  ] ConvolveTest2D/AV1HbdJntConvolve2DTest.MatchTest/0, where GetParam() = (8, 1, 1, 0, 0)
[  FAILED  ] ConvolveTest2D/AV1HbdJntConvolve2DTest.MatchTest/1, where GetParam() = (8, 1, 1, 0, 1)
[  FAILED  ] ConvolveTest2D/AV1HbdJntConvolve2DTest.MatchTest/16, where GetParam() = (8, 1, 1, 0, 16)
[  FAILED  ] ConvolveTest2D/AV1HbdJntConvolve2DTest.MatchTest/22, where GetParam() = (10, 1, 1, 0, 0)
[  FAILED  ] ConvolveTest2D/AV1HbdJntConvolve2DTest.MatchTest/23, where GetParam() = (10, 1, 1, 0, 1)
[  FAILED  ] ConvolveTest2D/AV1HbdJntConvolve2DTest.MatchTest/38, where GetParam() = (10, 1, 1, 0, 16)
[  FAILED  ] ConvolveTest2D/AV1HbdJntConvolve2DTest.MatchTest/44, where GetParam() = (12, 1, 1, 0, 0)
[  FAILED  ] ConvolveTest2D/AV1HbdJntConvolve2DTest.MatchTest/45, where GetParam() = (12, 1, 1, 0, 1)
[  FAILED  ] ConvolveTest2D/AV1HbdJntConvolve2DTest.MatchTest/60, where GetParam() = (12, 1, 1, 0, 16)
[  FAILED  ] ConvolveTestX/AV1HbdJntConvolve2DTest.MatchTest/0, where GetParam() = (8, 1, 0, 0, 0)
[  FAILED  ] ConvolveTestX/AV1HbdJntConvolve2DTest.MatchTest/1, where GetParam() = (8, 1, 0, 0, 1)
[  FAILED  ] ConvolveTestX/AV1HbdJntConvolve2DTest.MatchTest/16, where GetParam() = (8, 1, 0, 0, 16)
[  FAILED  ] ConvolveTestX/AV1HbdJntConvolve2DTest.MatchTest/22, where GetParam() = (10, 1, 0, 0, 0)
[  FAILED  ] ConvolveTestX/AV1HbdJntConvolve2DTest.MatchTest/23, where GetParam() = (10, 1, 0, 0, 1)
[  FAILED  ] ConvolveTestX/AV1HbdJntConvolve2DTest.MatchTest/38, where GetParam() = (10, 1, 0, 0, 16)
[  FAILED  ] ConvolveTestX/AV1HbdJntConvolve2DTest.MatchTest/44, where GetParam() = (12, 1, 0, 0, 0)
[  FAILED  ] ConvolveTestX/AV1HbdJntConvolve2DTest.MatchTest/45, where GetParam() = (12, 1, 0, 0, 1)
[  FAILED  ] ConvolveTestX/AV1HbdJntConvolve2DTest.MatchTest/60, where GetParam() = (12, 1, 0, 0, 16)
[  FAILED  ] ConvolveTestY/AV1HbdJntConvolve2DTest.MatchTest/0, where GetParam() = (8, 0, 1, 0, 0)
[  FAILED  ] ConvolveTestY/AV1HbdJntConvolve2DTest.MatchTest/1, where GetParam() = (8, 0, 1, 0, 1)
[  FAILED  ] ConvolveTestY/AV1HbdJntConvolve2DTest.MatchTest/16, where GetParam() = (8, 0, 1, 0, 16)
[  FAILED  ] ConvolveTestY/AV1HbdJntConvolve2DTest.MatchTest/22, where GetParam() = (10, 0, 1, 0, 0)
[  FAILED  ] ConvolveTestY/AV1HbdJntConvolve2DTest.MatchTest/23, where GetParam() = (10, 0, 1, 0, 1)
[  FAILED  ] ConvolveTestY/AV1HbdJntConvolve2DTest.MatchTest/38, where GetParam() = (10, 0, 1, 0, 16)
[  FAILED  ] ConvolveTestY/AV1HbdJntConvolve2DTest.MatchTest/44, where GetParam() = (12, 0, 1, 0, 0)
[  FAILED  ] ConvolveTestY/AV1HbdJntConvolve2DTest.MatchTest/45, where GetParam() = (12, 0, 1, 0, 1)
[  FAILED  ] ConvolveTestY/AV1HbdJntConvolve2DTest.MatchTest/60, where GetParam() = (12, 0, 1, 0, 16)
[  FAILED  ] AV1/AV1WienerConvolveHbdTest.random_test/3, where GetParam() = ((88, 88), 0x55c1fe12c640, 8)
[  FAILED  ] AV1/AV1WienerConvolveHbdTest.random_test/4, where GetParam() = ((88, 88), 0x55c1fe12c640, 10)
[  FAILED  ] AV1/AV1WienerConvolveHbdTest.random_test/5, where GetParam() = ((88, 88), 0x55c1fe12c640, 12)
[  FAILED  ] AV1/AV1WienerConvolveHbdTest.random_test/9, where GetParam() = ((72, 72), 0x55c1fe12c640, 8)
[  FAILED  ] AV1/AV1WienerConvolveHbdTest.random_test/10, where GetParam() = ((72, 72), 0x55c1fe12c640, 10)
[  FAILED  ] AV1/AV1WienerConvolveHbdTest.random_test/11, where GetParam() = ((72, 72), 0x55c1fe12c640, 12)
[  FAILED  ] AV1/AV1WienerConvolveHbdTest.random_test/15, where GetParam() = ((56, 56), 0x55c1fe12c640, 8)
[  FAILED  ] AV1/AV1WienerConvolveHbdTest.random_test/16, where GetParam() = ((56, 56), 0x55c1fe12c640, 10)
[  FAILED  ] AV1/AV1WienerConvolveHbdTest.random_test/17, where GetParam() = ((56, 56), 0x55c1fe12c640, 12)
[  FAILED  ] AV1/AV1WienerConvolveHbdTest.random_test/24, where GetParam() = ((24, 24), 0x55c1fe12c640, 8)
[  FAILED  ] AV1/AV1WienerConvolveHbdTest.random_test/25, where GetParam() = ((24, 24), 0x55c1fe12c640, 10)
[  FAILED  ] AV1/AV1WienerConvolveHbdTest.random_test/26, where GetParam() = ((24, 24), 0x55c1fe12c640, 12)
[  FAILED  ] AV1/AV1WienerConvolveHbdTest.random_test/30, where GetParam() = ((8, 8), 0x55c1fe12c640, 8)
[  FAILED  ] AV1/AV1WienerConvolveHbdTest.random_test/31, where GetParam() = ((8, 8), 0x55c1fe12c640, 10)
[  FAILED  ] AV1/AV1WienerConvolveHbdTest.random_test/32, where GetParam() = ((8, 8), 0x55c1fe12c640, 12)

Test cases running time sorting result as following:

|Test Case|Running Time|
-- | --
AV1/AV1WienerConvolveLbdTest | 2570239
C/AV1WarpFilterTest | 1209568
AVX2/AV1WarpFilterTest | 914162
LOOPSAD/sad_LoopTest | 818053
HMESAD/sad_LoopTest | 447610
Cdef/CDEFBlockTest | 259926
ConvolveTest2D/AV1HbdJntConvolve2DTest | 225230
EbRestorationPick_avx2 | 140569
BlendA64Mask1DTestHBD | 129564
BlendA64Mask1DTest8B | 128654
RST/PixelProjErrorLbdTest | 91128
BlendA64MaskTest8B | 79744
ConvolveTest2D/AV1LbdJntConvolve2DTest | 76212
SSE4_1/AV1HighbdWarpFilterTest | 50249
Av1KMeansDim/Av1KMeansDim | 46286
BlendA64MaskTestHBD_d16 | 39964
BlendA64MaskTestHBD | 38849
OBMC/OBMCSubPixelVarianceTest | 20304
BlendA64MaskTest8B_d16 | 20171
OBMC/OBMCsad_Test | 19968
OBMC/OBMCVarianceTest | 19830
WedgeUtilTest | 18266
SSE2/EncodeTxbTest | 16964
SelfGuidedToolsTest | 15068
ConvolveTest2D/AV1HbdSrConvolve2DTest | 11927
AV1/PsnrCalcHbdTest | 9250
TemporalFilter/TemporalFilterTest | 8147
BLEND/HbdCompBlendD16Test | 7246
BLEND/HbdCompBlendTest | 6950
AV1/AV1WienerConvolveHbdTest | 6673
AVCMCPPU/AVCStyleMcpPUTest | 6232
EIGHTSAD/GetEightsad_Test | 5495
PalleteMode/KMeansTest | 5129
BLEND/HbdCompBlendHMaskTest | 4869
BLEND/HbdCompBlendVMaskTest | 4772
Cdef/CDEFFindDirTest | 4679
BLEND/LbdCompBlendTest | 4511
SAD/SADTest | 4452
SSDAvg/SSDAvgTest | 4450
SAD/SADTestSubSample | 4422
SAD/SADAvgTest | 4420
ConvolveTest2D/AV1LbdSrConvolve2DTest | 4096
ConvolveTestX/AV1HbdJntConvolve2DTest | 3523
ConvolveTestY/AV1HbdJntConvolve2DTest | 3508
AVCMCPSearchRegion/AVCStyleMcpSearchRegionTest | 2932
BLEND/LbdCompBlendD16Test | 2302
GlobalMotion/RansacIntTest | 2179
GlobalMotion/RansacDoubleTest | 2068
TX/AV1FwdTxfm2dTest | 1879
UpsampleTestLBD | 1761
TX/InvTxfm2dAsmTest | 1683
TX_AVX2/InvTxfm2dAsmTest | 1678
BLEND/LbdCompBlendHMaskTest | 1621
BLEND/LbdCompBlendVMaskTest | 1591
IntraFilterEdgeTestHighbd | 1525
IntraFilterEdgeTestLowbd | 1492
ConvolveTestY/AV1LbdJntConvolve2DTest | 1193
ConvolveTestX/AV1LbdJntConvolve2DTest | 1192
RST/PixelProjErrorHbdTest | 855
intrapred/HighbdIntraPredTest | 816
intrapred/LowbdIntraPredTest | 740
TX/AV1FwdTxfm1dTest | 729
AVX2/AV1FrameErrorTest | 721
SAD/SADTestSubSample16bit | 633
ColorCountHbdTest | 627
SSE2/HbdLoopFilterTest | 622
FFT/FFT2DTest | 559
ALLSAD/Allsad_CalculationTest | 506
EXTSAD/Extsad_CalculationTest | 506
CALSAD/sad_CalculationTest | 505
AVX2/AV1HighbdSelfguidedFilterTest | 405
Variance/VarianceTest | 387
CdefToolTest | 382
Variance/HbdVarianceTest | 323
Quant/QuantizeBTest | 306
CompUtilTest/BuildCompDiffwtdMaskTest | 304
AV1/FilterIntraPredTest | 240
ConvolveTestY/AV1HbdSrConvolve2DTest | 240
ConvolveTestX/AV1HbdSrConvolve2DTest | 239
FullDistortionKernel16FuncTest/FullDistortionKernel16BitsFuncTest | 232
AV1/PsnrCalcLbdTest | 223
IFFT/IFFT2DTest | 217
ColorCountLbdTest | 189
AVCMCPSearchRegionHelper/AVCStyleMcpSearchRegionTestHelper | 181
AVCMCPPU_HELPER/AVCStyleMcpPUTestHelper | 177
UNPACKAVG/UnPackAvgTest | 139
AVX2/AV1SelfguidedFilterTest | 131
AVX2_COPY/AV1HbdJntConvolve2DTest | 114
SSE2/LbdLoopFilterTest | 112
ConvolveTestY/AV1LbdSrConvolve2DTest | 80
SpatialKernelFunc/SpatialFullDistortionKernelFuncTest | 79
ConvolveTestX/AV1LbdSrConvolve2DTest | 79
AVX2/CflSubAvgTest | 75
ComputeMeanTest | 74
TX/FwdTxfm2dAsmTest | 61
UNPACK/UnPackTest | 61
ConvolveTestCopy/AV1HbdSrConvolve2DTest | 61
Variance/HbdSquareVarianceTest | 43
LowbdDrZ2Test | 42
BitstreamWriterTest | 41
ConvolveTestCOPY/AV1LbdJntConvolve2DTest | 38
PACK2D/Pack2dTest | 36
SSE4_1/BuildCompDiffwtdMaskD16Test | 33
ExtractFilterTest | 32
HighbdDrZ2Test | 26
TX/AV1InvTxfm1dTest | 26
ConvolveTestCopy/AV1LbdSrConvolve2DTest | 21
ResidualUtil/ResidualKernelTest | 20
LowbdDrZ1Test | 17
LowbdDrZ3Test | 17
PictureOperator/PictureOperatorTest | 17
SpatialFunc/SpatialFullDistortionFuncTest | 17
HighbdDrZ1Test | 15
Entropy/EncodeTxbInitLevelTest | 14
HighbdDrZ3Test | 13
DenoiseModelRunTest | 12
HBDVarianceTest/HBDVarianceTest | 11
MotionEstimation_avx2 | 7
PACK/PackTest | 5
AVX2/SpatialFullDistortionTest | 5
Variance/MseTest | 5
Variance/MseTestHighbd | 5
HbdCflPredMatchTest | 3
AV1CornerMatchTest/AV1CornerMatchTest | 3
Compute8x8SatdTest | 2
LbdCflPredMatchTest | 2
PACKMSB/PackMsbTest | 1
AdaptiveScanTest | 0
Entropy_BitstreamWriter | 0
FilmGrain | 0
AddFilmGrainTest | 0
TransposeTest | 0
NNPRED/NNPredTest | 0
AV1/DvValiationTest | 0
InitializeBuffer32/InitializeBuffer32 | 0
TRANSPOSE/Transpose32Bit8x8Test | 0
TRANSPOSE/Transpose64BitTest | 0
Variance/SumSquareTest | 0

**Comment by Cidana-Emma** [comment]
Above failures are fixed by PR#954 in SVT-AV1 Git, so it is not necessary to submit the PR to disable the failed unit tests.
Checkout to-do task 1/2/3

**Comment by edmond-zhu** [comment]
Found segmentation fault in Travis CI test:
~~~
[----------] 1 test from TX_AVX2/InvTxfm2dAsmTest
[ RUN      ] TX_AVX2/InvTxfm2dAsmTest.rect_type1_txfm_match_test/0
/home/travis/.travis/functions: line 111: 15395 Segmentation fault      (core dumped) env GTEST_SHARD_INDEX=$SHARD_IDX SvtAv1UnitTests
The command "cd $TRAVIS_BUILD_DIR
export GTEST_TOTAL_SHARDS=$SHARDS
env GTEST_SHARD_INDEX=$SHARD_IDX SvtAv1UnitTests
unset GTEST_TOTAL_SHARDS
" exited with 0.
~~~

**[Done]** this issue is fixed in [Fix crash in transform unit tests](https://github.com/Cidana-Developers/SVT-AV1/pull/117)

**Comment by edmond-zhu** [comment]
New failure test cases found [HERE](https://travis-ci.com/Cidana-Developers/SVT-AV1/jobs/276545037):

Set Environment:
GTEST_TOTAL_SHARDS=8
GTEST_SHARD_INDEX=1

~~~
[==========] 3638 tests from 91 test cases ran. (1625994 ms total)
[  PASSED  ] 3630 tests.
[  FAILED  ] 8 tests, listed below:
[  FAILED  ] EXTSAD/Extsad_CalculationTest.Extsad_32x32Test/2, where GetParam() = (0, 2)
[  FAILED  ] EXTSAD/Extsad_CalculationTest.Extsad_32x32Test/10, where GetParam() = (2, 2)
[  FAILED  ] CALSAD/sad_CalculationTest.sad_32x32Test/2, where GetParam() = (0, 2)
[  FAILED  ] CALSAD/sad_CalculationTest.sad_32x32Test/10, where GetParam() = (2, 2)


[ RUN      ] EXTSAD/Extsad_CalculationTest.Extsad_32x32Test/6
/home/travis/build/Cidana-Developers/SVT-AV1/test/SadTest.cc:1654: Failure
Expected equality of these values:
  0
  memcmp( best_sad32x32[0], best_sad32x32[1], sizeof(best_sad32x32[0]))
    Which is: 167
compare best_sad32x32 error
[  FAILED  ] EXTSAD/Extsad_CalculationTest.Extsad_32x32Test/6, where GetParam() = (1, 2) (0 ms)
[ RUN      ] EXTSAD/Extsad_CalculationTest.Extsad_32x32Test/14
/home/travis/build/Cidana-Developers/SVT-AV1/test/SadTest.cc:1654: Failure
Expected equality of these values:
  0
  memcmp( best_sad32x32[0], best_sad32x32[1], sizeof(best_sad32x32[0]))
    Which is: 167
compare best_sad32x32 error
[  FAILED  ] EXTSAD/Extsad_CalculationTest.Extsad_32x32Test/14, where GetParam() = (3, 2) (0 ms)
[----------] 4 tests from EXTSAD/Extsad_CalculationTest (84 ms total)
~~~
[HERE:](https://travis-ci.com/Cidana-Developers/SVT-AV1/jobs/276580092)
~~~
[==========] 3574 tests from 84 test cases ran. (1269471 ms total)
[  PASSED  ] 3573 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] NNPRED/NNPredTest.NNPredTest/0, where GetParam() = 0x563e43ad4a60
~~~
@Richard-li  Would you please help check the root cause of these failure?

**Comment by Cidana-Emma** [comment]
1. PR#967: 
- Implement unit test in GitHub Actions with Shards based on PR#958, in order to workaround the long test time issue.
- Disable failed unit test cases
   a. Convolve and HBD Variance failed unit test cases: those failures are fixed by PR#954, but it hasn't been merged.
   b. SAD_Calculation: is fixed by PR#969.
- fixed the crash in Transform based on PR#944 to avoid being interrupted during testing.

2.  PR#969: fix SAD_Calculation failed test cases

**Comment by Cidana-Emma** [comment]
PRs have been merged, so close this issue
